autogluon_zeroshot/simulation/repository.py

```diff
@@ -141,12 +199,99 @@ def evaluate_ensemble(
             ] for fold in folds
         ] for dataset in dataset_names])
 
+    def get_datasets(self, problem_type=None):
```

Comment:
Given that there is a `dataset_names` method already, it would be nice to avoid adding one called `get_datasets`. We can rather have one, for instance `def dataset_names(self, problem_type: Optional[str] = None)` that returns the datasets of a specific type.

Reply:
`get_datasets` uses the inner value, which is the taskid. This aligns with the `self._zeroshot_context.get_datasets()` method.

I agree that we should try to unify, but I find myself always using the `taskid` representation rather than the string dataset name.

I'd recommend we defer this unification until later as it would warrant also refactoring zeroshot context, which is a lot more work. Currently it is rather easy to convert from one to the other, and I don't want to block the PR on this.